app/services/events/stores/postgres/unique_count_query.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Events
+  module Stores
+    module Postgres
+      class UniqueCountQuery
+        def initialize(store:)
+          @store = store
+        end
+
+        def query
+          # NOTE: First sum calculates all operation values for a specific property
+          # (for instance 2 relevant additions with 1 relevant removal [0, 1, 0, -1, 1] returns 1)
+          # The next sum combines all properties into a single result
+          <<-SQL
+            #{events_cte_sql},
+            event_values AS (
+              SELECT
+                property,
+                SUM(adjusted_value) AS sum_adjusted_value
+              FROM (
+                SELECT
+                  timestamp,
+                  property,
+                  operation_type,
+                  #{operation_value_sql} AS adjusted_value
+                FROM events_data
+                ORDER BY timestamp ASC
+              ) adjusted_event_values
+              GROUP BY property
+            )
+
+            SELECT SUM(sum_adjusted_value) AS aggregation FROM event_values
+          SQL
+        end
+
+        # NOTE: Not used in production, only for debug purpose to check the computed values before aggregation
+        # Returns an array of event's timestamp, property, operation type and operation value
+        # Example:
+        # [
+        #   ["2023-03-16T00:00:00.000Z", "001", "add", 1],
+        #   ["2023-03-17T00:00:00.000Z", "001", "add", 0],
+        #   ["2023-03-17T10:00:00.000Z", "002", "remove", 0],
+        #   ["2023-03-18T00:00:00.000Z", "001", "remove", -1],
+        #   ["2023-03-19T00:00:00.000Z", "002", "add", 1]
+        # ]
+        def breakdown_query
+          <<-SQL
+            #{events_cte_sql}
+
+            SELECT
+              timestamp,
+              property,
+              operation_type,
+              #{operation_value_sql}
+            FROM events_data
+            ORDER BY timestamp ASC
+          SQL
+        end
+
+        private
+
+        attr_reader :store
+
+        delegate :events, :sanitized_propery_name, to: :store
+
+        def events_cte_sql
+          # NOTE: Common table expression returning event's timestamp, property name and operation type.
+          <<-SQL
+            WITH events_data AS (
+              (#{
+                events
+                  .select(
+                    "timestamp, \
+                    #{sanitized_propery_name} AS property, \
+                    COALESCE(events.properties->>'operation_type', 'add') AS operation_type"
```

Comment:
Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)